agate/agate/tests/test_queue_reading.py

```diff
@@ -0,0 +1,154 @@
+from django.test import TestCase
+from unittest.mock import Mock
+from agate.queue_reading.queue_reader import QueueReader
+from agate.queue_reading.tracking_models import Project, ProjectSite
+from agate.models import IngestionAttempt
+from agate.i_message_retrieval import iMessageRetrieval
+
+inbound_matched_example = """{
+   "uuid":"4d35e1bf-44f9-4901-a8c6-b4751746a722",
+   "site":"bham",
+   "raw_site":"synthscape.ukhsa",
+   "uploaders":[
+      "bryn-synthscape-ukhsa"
+   ],
+   "match_timestamp":1727098664533980741,
+   "artifact":"synthscape|0|92274db4-3535-47db-b6f6-151354597142",
+   "run_index":"0",
+   "run_id":"92274db4-3535-47db-b6f6-151354597142",
+   "project":"synthscape",
+   "platform":"illumina",
+   "files":{
+      ".1.fastq.gz":{
+         "uri":"s3://synthscape-synthscape.ukhsa-illumina-test/synthscape.0.92274db4-3535-47db-b6f6-151354597142.1.fastq.gz",
+         "etag":"f667f435136e91b986254d735cee1c13",
+         "key":"synthscape.0.92274db4-3535-47db-b6f6-151354597142.1.fastq.gz",
+         "submitter":"bryn-synthscape-ukhsa",
+         "parsed_fname":{
+            "project":"synthscape",
+            "run_index":"0",
+            "run_id":"92274db4-3535-47db-b6f6-151354597142",
+            "direction":"1",
+            "ftype":"fastq",
+            "gzip":"gz"
+         }
+      },
+      ".2.fastq.gz":{
+         "uri":"s3://synthscape-synthscape.ukhsa-illumina-test/synthscape.0.92274db4-3535-47db-b6f6-151354597142.2.fastq.gz",
+         "etag":"de2fed484dcad68d50ec6673df6b3670",
+         "key":"synthscape.0.92274db4-3535-47db-b6f6-151354597142.2.fastq.gz",
+         "submitter":"bryn-synthscape-ukhsa",
+         "parsed_fname":{
+            "project":"synthscape",
+            "run_index":"0",
+            "run_id":"92274db4-3535-47db-b6f6-151354597142",
+            "direction":"2",
+            "ftype":"fastq",
+            "gzip":"gz"
+         }
+      },
+      ".csv":{
+         "uri":"s3://synthscape-synthscape.ukhsa-illumina-test/synthscape.0.92274db4-3535-47db-b6f6-151354597142.csv",
+         "etag":"714c18df9c62b30f70443219467fcba1",
+         "key":"synthscape.0.92274db4-3535-47db-b6f6-151354597142.csv",
+         "submitter":"bryn-synthscape-ukhsa",
+         "parsed_fname":{
+            "project":"synthscape",
+            "run_index":"0",
+            "run_id":"92274db4-3535-47db-b6f6-151354597142",
+            "ftype":"csv"
+         }
+      }
+   },
+   "test_flag":true
+}"""
+
+
+class QueueReadingTestCase(TestCase):
+
+    def test_calling_inbound_matched(self):
+
+        self.assertEqual(IngestionAttempt.objects.count(), 0)
+
+        retrieval = iMessageRetrieval()
+        retrieval.receive_batch = Mock(return_value=[])
```

Comment:
Is this different from using `EmptyMessageRetrieval`?

Reply:
Yes, the mock object can count how many times the method has been called and with what arguments. This is later used for the assertion on line 79. This tests whether the method is doing what we expect.